### PR TITLE
docs: revamp

### PR DESCRIPTION
**Pull request by mdelapenya**
- docs: fix typo
- docs: fix typo
- docs: fix typo
- docs: add issue form for new platform support
- docs: typo
- docs: revamp main readme
- docs: add how to run the k8s-autodiscover test suite for a commit
- docs: refinements in fleet docs

## What does this PR do?
It adds the content of an email thread to the main readme file for the e2e project, fixing a few typos and finally adding an issue template with a [github form](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) to request a new platform support.

## Why is it important?
Keep docs more usable

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

-

### CODE REVIEW

File: e2e/_suites/kubernetes-autodiscover/README.md [hunk 1]
```diff
@@ -132,6 +132,54 @@ This is an example of the optional configuration:
...
+   - Install Kind 0.10.0 or newer
+   - Install Go, using the language version defined in the `.go-version` file at the root directory. We recommend using [GVM](https://github.com/andrewkroh/gvm), same as done in the CI, which will allow you to install multiple versions of Go, setting the Go environment in consequence: `eval "$(gvm 1.15.9)"`
+   - Godog and other test-related binaries will be installed in their supported versions when the project is first built, thanks to Go modules and Go build system.
+
+4. Verify that the elastic-agent or beats binaries have been published by the CI in the right GCP bucket. You can look up the bucket and identify the PR identifier (ex. *BUCKET_URI/pull-requests/PR-123*) or the commit hash of the branch that was already built by the CI (ex. *$BUCKET_URI/commits/SHA*).
+
+5. Run the tests for the commit.
+   ```shell
+   cd e2e/_suites/kubernetes-autodiscover
+   export GITHUB_CHECK_SHA=sha
```
Comment (mdelapenya) [inline e2e/_suites/kubernetes-autodiscover/README.md:167]: @jsoriano @ChrsMark pinging you for awareness in this section of the refactor

File: e2e/tooling.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Tooling
```
Comment (mdelapenya) [inline e2e/tooling.md:1]: Extracted from the main README.md file

File: e2e/_suites/kubernetes-autodiscover/README.md (hunk 1 above)
Comment (jsoriano) [reply e2e/_suites/kubernetes-autodiscover/README.md:167]: Thanks for the ping.

Points 1 to 4 are the same in the previous section, could we make this section a subsection of the previous one, adding here the additional steps?

Comment (mdelapenya) [reply e2e/_suites/kubernetes-autodiscover/README.md]: Done in https://github.com/elastic/e2e-testing/pull/2266/commits/021fbb58df3e10cdf5f766616af0b7b4b4d4291f

File: e2e/README.md [hunk 3]
```diff
@@ -1,92 +1,62 @@
...
+  - Julia Bardi and Nicolas Chaulet, from Fleet team, frontend engineers, have contributed a few scenarios for Fleet
+  - Eric Davis, QA engineer, has helped in the definition of the scenarios for Fleet
+  - Igor Guz, from Security team, QA engineer, has contributed scenarios for the security-related integrations, such as Endpoint, Linux and System
+  - Christos Markou and Jaime Soriano, from Beats, have contributed and currently maintain the entire k8s-autodiscover test suite
+  - Julien Lind, from Fleet, has helped in defining the support matrix in terms of what OSs and architectures need to be run for Fleet test suite
+  - Julien Mailleret, from Infra, has contributed to the Helm charts test suite.
+  - Anderson Queiroz, from Elastic Agent, is currently working on the MacOS support for running the tests on real Apple machines.
```
Comment (cachedout) [inline e2e/README.md:56]: Eh? I just talked to @adam-stokes yesterday and I was left with the impression that this was still a TODO for him to do and that we were going to need to integrate this with the upcoming Orka integration, where we're already trying to estimate the total number of machines we might need. Could you please link me to an issue or issues where this support is being discussed?

File: e2e/README.md [hunk 4]
```diff
@@ -1,92 +1,62 @@
...
 
-For this test framework, we have chosen Godog over any other test framework because the Beats team (the team we started working with) is already using Go, so it seems reasonable to choose it.
+- The target audience is matching the aforementioned suites, although any other team can come and add their suites.
 
-## Technology stack
+## CI execution
+On the CI, the framework creates a VM for the Stack (Elasticsearch, Kibana and Fleet Server) first, using the current valid SNAPSHOT. It's valid because we have validated in a pull-request that a given snapshot is not breaking the tests. This process is automated: everyday, we receive an automated bump of the different snapshots for all the supported maintenance branches (main, 8.1, 8.0 and 7.17 as of today).
+  - In the fleet test suite, Fleet Server is part of the Stack. The scenarios are executed connecting the VMs where the elastic-agent is installed to the Stack machine.
+  - A pull-request in the elastic-agent repo runs the Fleet scenarios using the CI snapshots for that PR for the agent, and the stack version int the target maintenance branch for the beats in play (filebeat, heartbeat, metricbeat)
```
[the remark this answers is not in the document]
Comment (mdelapenya) [reply e2e/README.md:38]: It's the target branch for the pull-request, at this moment: main, 7.17, 8.0, 8.1. Maybe rewording it to `in the target branch of the pull request for the beats in play`?

File: e2e/README.md [hunk 5]
```diff
@@ -1,92 +1,62 @@
...
+  - `Then` an outcome is expected (mandatory)
+  - "And/But" clauses are allowed to provide continuation on the above ones. (Optional)
+- Because it uses BDD, it's possible to combine existing scenario steps (the given-when-then clauses) forming new scenarios, so with no code it's possible to have new tests. For that reason, the steps must be atomic and decoupled (as any piece of software: low coupling and high cohesion).
+- It does not use the GUI at all, so no Selenium, Cypress or any other test library having expectations on graphical components. It uses Go as the programming language and Cucumber to create the glue code between the code and the plain English scenarios. In almost two years, we have demonstrated that the APIs are not changing as fast as the GUI components.
+- APIs not changing does not mean zero flakiness. Because there are so many moving pieces (stack versions), beats versions, elastic-agent versions, cloud machines, network access, etc... There could be situations where the tests fail, but they are rarely caused by test flakiness. Instead, they are usually caused by: 1) instability of the all-together system, 2) a real bug, 3) a badly written test.
```
[the remark this answers is not in the document]
Comment (mdelapenya) [reply e2e/README.md:21]: I wanted to express the inconsistencies caused by gherkin steps that are not consistent. Maybe replacing it like that could help.

File: e2e/README.md (hunk 4 above)
Comment (mdelapenya) [inline e2e/README.md:38]:
```suggestion
  - A pull-request in the elastic-agent repo runs the Fleet scenarios using the CI snapshots for that PR for the agent, and the stack version in the target branch of the pull request for the beats in play (filebeat, heartbeat, metricbeat)
```

File: e2e/README.md [hunk 6]
```diff
@@ -1,92 +1,62 @@
...
+- The target audience is matching the aforementioned suites, although any other team can come and add their suites.
 
-## Technology stack
+## CI execution
+On the CI, the framework creates a VM for the Stack (Elasticsearch, Kibana and Fleet Server) first, using the current valid SNAPSHOT. It's valid because we have validated in a pull-request that a given snapshot is not breaking the tests. This process is automated: everyday, we receive an automated bump of the different snapshots for all the supported maintenance branches (main, 8.1, 8.0 and 7.17 as of today).
+  - In the fleet test suite, Fleet Server is part of the Stack. The scenarios are executed connecting the VMs where the elastic-agent is installed to the Stack machine.
+  - A pull-request in the elastic-agent repo runs the Fleet scenarios using the CI snapshots for that PR for the agent, and the stack version int the target maintenance branch for the beats in play (filebeat, heartbeat, metricbeat)
+  - A pull-request in Beats repo runs the Fleet scenarios using the CI snapshots for that PR for the beats, and the stack version int the target maintenance branch for the elastic-agent
```
Comment (mdelapenya) [inline e2e/README.md:39]:
```suggestion
  - A pull-request in Beats repo runs the Fleet scenarios using the CI snapshots for that PR for the beats, and the stack version in the target branch of the pull request for the elastic-agent.
```

File: e2e/README.md (hunk 5 above)
Comment (mdelapenya) [inline e2e/README.md:21]:
```suggestion
- APIs not changing does not mean zero flakiness. Because there are so many moving pieces (stack versions), beats versions, elastic-agent versions, cloud machines, network access, etc... There could be situations where the tests fail, but they are rarely caused by test flakiness. Instead, they are usually caused by: 1) instability of the all-together system, 2) a real bug, 3) gherkin steps that are not consistent.
```

File: e2e/README.md (hunk 3 above)
Comment (mdelapenya) [reply e2e/README.md:56]: I think we can merge this one even without the feedback from orka support

**Review by cachedout — APPROVED** [review]

**Review by jsoriano — APPROVED** [review]

File: e2e/README.md [hunk 7]
```diff
@@ -1,92 +1,62 @@
...
 
-### Runtime dependencies
-In many cases, we want to store the metrics in Elasticsearch, so at some point we must start up an Elasticsearch instance. Besides that, we want to query the Elasticsearch to perform assertions on the metrics, such as there are no errors, or the field `f.foo` takes the value `bar`. For that reason we need an Elasticsearch instance in a well-known location. We are going to group this Elasticsearch instance, and any other runtime dependencies, under the concept of a `profile`, which is a represented by a `docker-compose.yml` file under the `cli/config/compose/profiles/` and the name of the test suite.
+  - Julia Bardi and Nicolas Chaulet, from Fleet team, frontend engineers, have contributed a few scenarios for Fleet
+  - Eric Davis, QA engineer, has helped in the definition of the scenarios for Fleet
+  - Igor Guz, from Security team, QA engineer, has contributed scenarios for the security-related integrations, such as Endpoint, Linux and System
+  - Christos Markou and Jaime Soriano, from Beats, have contributed and currently maintain the entire k8s-autodiscover test suite
```
Comment (jsoriano) [inline e2e/README.md:53]: Well, the truth is that this part is currently maintained by @elastic/obs-cloudnative-monitoring :innocent:

Comment (mdelapenya) [reply e2e/README.md:53]: Let's change it then 😇

Comment (mdelapenya) [inline e2e/README.md:53]:
```suggestion
  - Christos Markou and Jaime Soriano have contributed to the k8s-autodiscover test suite, which is maintained by @elastic/obs-cloudnative-monitoring.
```
